src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -68,8 +68,9 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
     readonly isCommunicationEnabled = isCommunicationEnabled;
     readonly isMessagingEnabled = isMessagingEnabled;
 
+    public learningPathMode = false;
     private currentUser: User;
-    private exerciseId: number;
+    public exerciseId: number;
```

Comment:
Why do we need to make this public?

Reply:
I have to set it in `LearningPathContainerComponent` when the exercise details get routed (similar to what we do for conversations). Otherwise, the exercisId would be taken from the URL which is not possible since we don't actually want to navigate to a sub-route.